code/datums/ai/generic_actions.dm

```diff
@@ -22,3 +22,68 @@
 /datum/ai_behavior/move_to_target/perform(delta_time, datum/ai_controller/controller)
 	. = ..()
 	finish_action(controller, TRUE)
+
+
+///Finds an item near themselves, sets a blackboard key as it. Very useful for ais that need to use machines or something.
+/datum/ai_behavior/find_and_set
+	action_cooldown = 15 SECONDS
```

Comment:
bit worried about spaztics making 50 of these but we'll see in practice, so far the ai subsystem has been cheap.

Reply:
👍